### PR TITLE
feat(uma): discovery & protection api

### PR DESCRIPTION
**Pull request by lusoalex**

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/uma/resources/handler/MethodNotSupportedHandler.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * and MAY respond with an unsupported_method_type error code.
+ *
+ * @author Alexandre FARIA (contact at alexandrefaria.net)
+ * @author GraviteeSource Team
+ */
+public class MethodNotSupportedHandler implements Handler<RoutingContext> {
+
+    @Override
+    public void handle(RoutingContext context) {
+        context.fail(new OAuth2Exception() {
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/uma/resources/handler/MethodNotSupportedHandler.java:35]: Use a dedicated `MethodNotSupportException`

Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/uma/resources/handler/MethodNotSupportedHandler.java:35]: Use a dedicated UnsupportedMethodTypeException

Comment (lusoalex) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/uma/resources/handler/MethodNotSupportedHandler.java:35]: Got it, will ping you once done!

Comment (lusoalex) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/uma/resources/handler/MethodNotSupportedHandler.java]: @tcompiegne please check. Thanks!

**Pull request merged**